includes/tax-functions.php

```diff
@@ -453,22 +446,24 @@ function edd_get_tax_rate_by_location( $args ) {
 		);
 	}
 
-	// If no country specific tax rates are found, check for a global tax rate.
-	if ( empty( $tax_rates ) ) {
-		$tax_rates = edd_get_tax_rates(
-			array(
-				'name'   => '',
-				'scope'  => 'global',
-				'status' => 'active',
-			),
-			OBJECT
-		);
-	}
+	$fallback = edd_get_tax_rates(
+		array(
+			'name'   => '',
+			'scope'  => 'global',
+			'status' => 'active',
+		),
+		OBJECT
+	);
 
-	if ( empty( $tax_rates ) ) {
+	if ( empty( $tax_rates ) && empty( $fallback ) ) {
```

Comment:
Really, if we removed this early return and only got the fallback if `$rate` is still false after going through the `foreach`, that might save us a database call.